### PR TITLE
Revert v8 to 12.9

### PR DESCRIPTION
**Pull request by danlapid**
Revert "Merge pull request #2780 from cloudflare/dominik/v8-13.0"
Update v8 to latest version of 12.9 - 12.9.202.27

### CODE REVIEW

**Review by erikcorry — APPROVED** [review]

File: patches/v8/0004-Allow-Windows-builds-under-Bazel.patch [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
-index 1f269f9c149e56a7098697fc2346b5998b0c6e01..89bc2f54fc5e42a2cf6d6fa41742c2dfe2c9ab48 100644
+index f2b2f4da0f8164ef29e36a3d1e9489b8e9c7b86e..84a281458aaa18525fa3640b3710414153e008b1 100644
 --- a/BUILD.bazel
 +++ b/BUILD.bazel
-@@ -817,6 +817,7 @@ filegroup(
+@@ -820,6 +820,7 @@ filegroup(
          "src/base/platform/mutex.h",
          "src/base/platform/platform.cc",
          "src/base/platform/platform.h",
 +        "src/base/platform/platform-posix.h",  # Always included by src/execution/isolate.h
```
Comment (vickykont) [inline patches/v8/0004-Allow-Windows-builds-under-Bazel.patch:15]: If I'm not mistaken, it looks like this change was introduced in the 13.0 update. Same with the removed line below. So this doesn't look like a reverted file. Could you cross-check? Maybe I'm not reading this correctly.

Comment (danlapid) [reply patches/v8/0004-Allow-Windows-builds-under-Bazel.patch:15]: It's just a revert, you can look at the PR commit by commit to verify, the first commit is the revert commit

File: patches/v8/0007-Implement-Promise-Context-Tagging.patch [hunk 2]
```diff
@@ -126,7 +126,7 @@ index f4e17456651516dfb3e81880da82ab228646ccfd..6142e0450dac646ccebd3fc7c5168f9b
 -  promise.SetHasHandler();
```
Comment (vickykont) [inline patches/v8/0007-Implement-Promise-Context-Tagging.patch:126]: This one too.

File: patches/v8/0019-Implement-cross-request-context-promise-resolve-hand.patch [hunk 3]
```diff
@@ -1,27 +1,11 @@
-From 8cf4fe0f72fbaf25028ebd152559bd363da6599e Mon Sep 17 00:00:00 2001
+From 7724507f0525deef0a484d33d2efee3586728520 Mon Sep 17 00:00:00 2001
 From: James M Snell <jasnell@gmail.com>
 Date: Mon, 16 Sep 2024 09:56:04 -0700
-Subject: [PATCH] Implement cross-request context promise resolve handling
+Subject: Implement cross-request context promise resolve handling
 
----
```
Comment (vickykont) [inline patches/v8/0019-Implement-cross-request-context-promise-resolve-hand.patch]: Do we know why this patch is changed as part of this commit (the commit updating to the latest 12.9)?

**Review by vickykont — APPROVED** [review]

File: patches/v8/0004-Allow-Windows-builds-under-Bazel.patch (hunk 1 above)
Comment (erikcorry) [reply patches/v8/0004-Allow-Windows-builds-under-Bazel.patch:15]: Dan points out that this is github miscolouring the patch file.  This line was in the patch file all the time. Does that answer your question or is it still suspicious?

**Pull request merged**